### PR TITLE
Track all subnet validator sets in the validator manager

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

This improves the performance of warp and unblocks subnet specific logic in the networking layer.

## How this works

Modified the P-chain state to keep all subnets in the validator manager. This is not a significant increase in memory pressure because we already pin validators + delegators in memory on the P-chain.

## How this was tested

1. Locally verified that non-tracked subnets are now sample-able over the P-chain API (used WAGMI).
2. Updated the unit tests
3. CI

### CODE REVIEW

File: vms/platformvm/validators/manager.go
Comment (ceyonur) [inline vms/platformvm/validators/manager.go:1]: Do we also want to enable caching validators sets here? https://github.com/ava-labs/avalanchego/blob/20f3580b4912f29adc602ad883e84897f0965cc1/vms/platformvm/validators/manager.go#L229

**Review by joshua-kim — APPROVED** [review]

**Review by joshua-kim — APPROVED** [review]

File: vms/platformvm/validators/manager.go
Comment (StephenButtolph) [reply vms/platformvm/validators/manager.go:1]: We'll want to change how the caching works here in a followup - but just removing this check isn't sufficient

**Review by danlaine — APPROVED** [review]

File: vms/platformvm/state/state.go [hunk 1]
```diff
@@ -1689,38 +1660,36 @@ func (s *state) loadPendingValidators() error {
...
-	s.cfg.Validators.RegisterCallbackListener(constants.PrimaryNetworkID, vl)
+		for nodeID, validator := range validators {
+			staker := validator.validator
+			if err := s.cfg.Validators.AddStaker(subnetID, nodeID, staker.PublicKey, staker.TxID, staker.Weight); err != nil {
+				return err
+			}
+
+			delegatorIterator := NewTreeIterator(validator.delegators)
+			for delegatorIterator.Next() {
+				staker := delegatorIterator.Value()
```
Comment (patrick-ogrady) [inline vms/platformvm/state/state.go:1677]: nit: prefer to give this a different variable name

File: vms/platformvm/state/state.go [hunk 2]
```diff
@@ -1689,38 +1660,36 @@ func (s *state) loadPendingValidators() error {
...
+				if err := s.cfg.Validators.AddWeight(subnetID, nodeID, staker.Weight); err != nil {
+					delegatorIterator.Release()
+					return err
+				}
+			}
+			delegatorIterator.Release()
```
Comment (patrick-ogrady) [inline vms/platformvm/state/state.go:1683]: do we need to check some sort of `delegatorIterator.Error` here?

File: vms/platformvm/state/state.go (hunk 1 above)
Comment (patrick-ogrady) [reply vms/platformvm/state/state.go:1677]: I always end up causing a bug at some point in the future when I do nested for loops with the same var name in them

File: vms/platformvm/state/state.go (hunk 2 above)
Comment (StephenButtolph) [reply vms/platformvm/state/state.go:1683]: There is no `.Error()` to check for on staker iterators

**Pull request merged**